### PR TITLE
feat(cloud): send reviewers in deployment creation.

### PR DESCRIPTION
**Pull request by i4ki**
## What this PR does / why we need it:

The Terramate Cloud frontend needs the list of reviewers of each deployment.

## Special notes for your reviewer:

Unfortunately no tests were added because it requires a GH API fake server.

## Does this PR introduce a user-facing change?
```
no
```

### CODE REVIEW

**Review by wmalik — DISMISSED** [review]
Looks good to me (I understand that we don't have a nice way to test external services at the moment but still curious how we can manually test this).

File: cmd/terramate/cli/github/client.go [hunk 1]
```diff
@@ -91,6 +93,44 @@ func (c *Client) Commit(ctx context.Context, repository, sha string) (*Commit, e
 	return &commit, nil
 }
 
+// PullReviews returns the list of reviews of the given pull request number.
+func (c *Client) PullReviews(ctx context.Context, repository string, nr int) (Reviews, error) {
+	if err := validateRepo(repository); err != nil {
+		return nil, err
+	}
+	const perPage = 100
+	urlStr := fmt.Sprintf("%s/repos/%s/pulls/%d/reviews", c.baseURL(), repository, nr)
```
Comment (sixstone-qq) [inline cmd/terramate/cli/github/client.go:102]: I thought you wanted to get the list of reviewers from the ones that was requested (which also includes the ones that made reviews which returns this endpoint). I mean `requested_reviewers` field in https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#get-a-pull-request

File: cloud/types.go [hunk 2]
```diff
@@ -220,15 +220,26 @@ type (
...
+		Description string    `json:"description"`
+		URL         string    `json:"url"`
+		Reviewers   Reviewers `json:"reviewers,omitempty"`
 	}
 
+	// Reviewer is the user's reviewer of a Pull/Merge Request.
+	Reviewer struct {
+		Login      string `json:"login"`
+		AvatarURL  string `json:"avatar_url,omitempty"`
+		GravatarID string `json:"gravatar_id,omitempty"`
```
Comment (sixstone-qq) [inline cloud/types.go:237]: Do we really need this field? See https://developer.github.com/changes/2014-09-05-removing-gravatar-id/ and issues users have with this field https://github.com/orgs/community/discussions/38110

**Review by sixstone-qq — COMMENTED** [review]
Let me know what you think of my comments, I think they are relevant

File: cloud/types.go (hunk 2 above)
Comment (i4ki) [reply cloud/types.go:237]: Thanks for checking this! I will check if really needed.

File: cmd/terramate/cli/github/client.go (hunk 1 above)
Comment (i4ki) [reply cmd/terramate/cli/github/client.go:102]: The `requested_reviewers` becomes empty after a review is submitted so by the time that PR is ready to be merged (or changes applied to the IaC) then it contains an empty list.

Comment (i4ki) [reply cmd/terramate/cli/github/client.go:102]: Have a look here: https://api.github.com/repos/terramate-io/terramate/pulls/1175
This PR was reviewed.

Comment (sixstone-qq) [reply cmd/terramate/cli/github/client.go:102]: Awesome! That's great to know :people_hugging:

**Review by wmalik — APPROVED** [review]

**Pull request merged**